.platform/hooks/predeploy/02_fetch_ssh_keys.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Get the env type from EB
+ENV_TYPE=$(/opt/elasticbeanstalk/bin/get-config environment -k SSM_PREFIX)
+
+SSH_PUBLIC_KEY_PARAM_NAME="${ENV_TYPE}_SSH_PUBLIC_KEY"
+SSH_PRIVATE_KEY_PARAM_NAME="${ENV_TYPE}_SSH_PRIVATE_KEY"
+
+echo "Set AWS region"
+aws configure set default.region ap-southeast-1
+
+echo "Fetching keys"
+# Note we write to webapp user directory which runs our app
+aws ssm get-parameter --name $SSH_PUBLIC_KEY_PARAM_NAME --with-decryption --query "Parameter.Value" --output text > /home/webapp/.ssh/github.pub
+aws ssm get-parameter --name $SSH_PRIVATE_KEY_PARAM_NAME --with-decryption --query "Parameter.Value" --output text > /home/webapp/.ssh/github
+
+# Set the permissions for the keys
+echo "Setting permissions"
+chmod 600 /home/webapp/.ssh/github.pub
+chmod 600 /home/webapp/.ssh/github
```

Comment:
Since the files are created by `root` user, we also need to change the ownership of the files so the permissions would work for `webapp` user. Otherwise, `webapp` user may not be able to access them, right? 
we can use `chown` to change the ownership of the files.

Reply:
Yes we do that but at the end of all ssh operations. See script 4, we do a recursive owner change

```
# all the above would have been performed as root
# so we need to allow webapp user to access so that it can add to the known_hosts etc.
chown -R webapp:webapp /home/webapp/.ssh
```